File: in_clojure/day_01_part_1.clj
#!/usr/bin/env bb

(require '[clojure.string :as str])

(def input
  (->> "../inputs/day01.txt"
       (slurp)
       (str/split-lines)
       (map read-string)
       set))

(def solution
  (some identity
        (for [a input]
          (when-let [b ((disj input a) (- 2020 a))]
            (* a b)))))

(println "Day 01 - Part 1:" solution)

File: in_ruby/day_01_part_1.rb
require 'set'

input = File.read("../inputs/day01.txt").lines.map(&:to_i).to_set

solution = nil
input.each do |a|
  if input.include?(2020 - a)
    solution = a * (2020 - a)
    break
  end
end

#--
puts "Day 01 - Part 1: #{solution}"
